deps/rabbitmq_management_agent/src/rabbit_mgmt_metrics_gc.erl

```diff
@@ -137,18 +140,23 @@ delete_samples(Table, Id, Intervals) ->
     [ets:delete(Table, {Id, I}) || I <- Intervals],
     ok.
 
-index_delete(consumer_stats = Table, channel = Type, Id) ->
-    IndexTable = rabbit_mgmt_metrics_collector:index_table(Table, Type),
-    MatchPattern = {'_', Id, '_'},
-    %% Delete consumer_stats_queue_index
-    ets:match_delete(consumer_stats_queue_index,
-                     {'_', MatchPattern}),
-    %% Delete consumer_stats
-    ets:match_delete(consumer_stats,
-                     {MatchPattern,'_'}),
-    %% Delete consumer_stats_channel_index
-    ets:delete(IndexTable, Id),
-    ok;
+index_delete(consumer_stats = Table, {channel = Type, Count}, Id) ->
+    case Count > ?LARGE_CONSUMER_COUNT of
+        true ->
+            IndexTable = rabbit_mgmt_metrics_collector:index_table(Table, Type),
+            MatchPattern = {'_', Id, '_'},
+            %% Delete consumer_stats_queue_index
+            ets:match_delete(consumer_stats_queue_index,
+                             {'_', MatchPattern}),
+            %% Delete consumer_stats
+            ets:match_delete(consumer_stats,
+                             {MatchPattern,'_'}),
+            %% Delete consumer_stats_channel_index
+            ets:delete(IndexTable, Id),
+            ok;
+        false ->
+            index_delete(Table, Type, Id)
+    end;
 index_delete(Table, Type, Id) ->
```

Comment:
Just for completeness' sake, if a node gets an event that does not have the 2nd argument as a tuple, this function head should handle that. E.g. in a mixed version cluster situation during an ongoing upgrade.

Reply:
Hm, but if it does not have a second argument, i.e Type = consumer, line 160 should handle it, the way it did before I introduced the match_delete logic?

Reply:
@michaelklishin fixed the var names and comment. I think I am missing something here though!

Reply:
Yes, that's what I was referring to. Like 160 would handle the case without the consumer count :)